sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala

```diff
@@ -235,8 +236,17 @@ class ResolveSessionCatalog(
     case DescribeRelation(ResolvedV1TableOrViewIdentifier(ident), partitionSpec, isExtended) =>
       DescribeTableCommand(ident.asTableIdentifier, partitionSpec, isExtended)
 
-    case DescribeColumn(ResolvedV1TableOrViewIdentifier(ident), colNameParts, isExtended) =>
-      DescribeColumnCommand(ident.asTableIdentifier, colNameParts, isExtended)
+    case DescribeColumn(ResolvedV1TableOrViewIdentifier(ident), column, isExtended) =>
+      column match {
+        case u: UnresolvedAttribute =>
+          // For views, the column will not be resolved by `ResolveReferences` because
+          // `ResolvedView` stores only the identifier.
+          DescribeColumnCommand(ident.asTableIdentifier, u.nameParts, isExtended)
+        case a: Attribute =>
+          DescribeColumnCommand(ident.asTableIdentifier, a.qualifier :+ a.name, isExtended)
+        case nested =>
+          throw QueryCompilationErrors.commandNotSupportNestedColumnError("DESC TABLE COLUMN")
```

Comment:
One disadvantage of this approach is that the exception message for view will be different when nested column is specified; it will have the original name parts: https://github.com/apache/spark/blob/0617dfce7beb34662ab30a607721e9b46e65c21e/sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala#L773

We can do one of the following:
* Make the exception message the same even for views by dropping column name in `DescribeColumnCommand`
* Store the original column name in `DescribeColumn` (and there will be no matching logic for column in `ResolveSessionCatalog`, but seems duplicated because we have `UnresolvedAttribute` to store the original column name.)
* Construct the original name from `GetStructField`, `GetArrayStructFields`, etc.

WDYT, @cloud-fan ?

Reply:
> Construct the original name from GetStructField, GetArrayStructFields, etc.

Is it simply `nested.sql`?

Reply:
For `DESC desc_complex_col_table col.x`,
It will be:
```
DESC TABLE COLUMN command does not support nested data types: col.x
```
vs.
```
DESC TABLE COLUMN does not support nested column: spark_catalog.default.desc_complex_col_table.`col`.`x` AS `x`
```

Reply:
How about we strip the `Alias` and then call `toPrettySQL` which is defined in `org.apache.spark.sql.catalyst.util`?

Reply:
That seems to work pretty well. Thanks!